src/admin/resolvers/queries/ApprovedItem.auth.integration.ts

```diff
@@ -0,0 +1,264 @@
+      const headers = {
+        name: 'Test User',
+        username: 'test.user@test.com',
+        groups: `group1,group2,${MozillaAccessGroup.READONLY}`,
+      };
+
+      const server = getServerWithMockedHeaders(headers);
+      await server.start();
+
+      const result = await server.executeOperation({
+        query: GET_APPROVED_ITEMS,
+      });
+
+      expect(result.errors).to.be.undefined;
+      expect(result.data).not.to.be.null;
+
+      // should return all 3 items
+      expect(result.data?.getApprovedCuratedCorpusItems.edges).to.have.length(
+        3
+      );
+
+      await server.stop();
+    });
+
+    it('should get all items when user has only one scheduled surface access', async () => {
+      // Set up auth headers with access to a single Scheduled Surface
+      const headers = {
+        name: 'Test User',
+        username: 'test.user@test.com',
+        groups: `group1,group2,${MozillaAccessGroup.NEW_TAB_CURATOR_ENUS}`,
+      };
+
+      const server = getServerWithMockedHeaders(headers);
+      await server.start();
+
+      const result = await server.executeOperation({
+        query: GET_APPROVED_ITEMS,
+      });
+
+      expect(result.errors).to.be.undefined;
+      expect(result.data).not.to.be.null;
+
+      // should return all 3 items
+      expect(result.data?.getApprovedCuratedCorpusItems.edges).to.have.length(
+        3
+      );
+
+      await server.stop();
+    });
+
+    it('should throw an error when user does not have the required access', async () => {
+      // Set up auth headers with no valid access group
+      const headers = {
+        name: 'Test User',
+        username: 'test.user@test.com',
+        groups: `group1,group2`,
+      };
+
+      const server = getServerWithMockedHeaders(headers);
+      await server.start();
+
+      const result = await server.executeOperation({
+        query: GET_APPROVED_ITEMS,
+      });
+
+      expect(result.data).to.be.null;
+      expect(result.errors).not.to.be.undefined;
+
+      // check if the error we get is access denied error
+      expect(result.errors?.[0].message).to.equal(ACCESS_DENIED_ERROR);
+
+      await server.stop();
+    });
+
+    it('should throw an error when request header groups are undefined', async () => {
+      // Set up auth headers with no valid access group
+      const headers = {
+        name: 'Test User',
+        username: 'test.user@test.com',
+        groups: undefined,
+      };
+
+      const server = getServerWithMockedHeaders(headers);
+      await server.start();
+
+      const result = await server.executeOperation({
+        query: GET_APPROVED_ITEMS,
+      });
+
+      expect(result.data).to.be.null;
+      expect(result.errors).not.to.be.undefined;
+
+      // check if the error we get is access denied error
+      expect(result.errors?.[0].message).to.equal(ACCESS_DENIED_ERROR);
+
+      await server.stop();
+    });
+  });
+
+  describe('getApprovedCuratedCorpusItemByUrl query', () => {
```

Comment:
Tests for this query start from here. The ones above are simply moved to this file.